packages/angular/src/migrations/update-15-9-0/update-file-server-executor.ts

```diff
@@ -32,21 +34,17 @@ export default async function updateFileServerExecutor(tree: Tree) {
   }
 
   const projects = getProjects(tree);
-  for (const [projectName, project] of projects.entries()) {
-    let projectChanged = false;
-
-    for (const [targetName, target] of Object.entries(project.targets)) {
-      if (target.executor === oldExecutor) {
-        project.targets[targetName].executor = newExecutor;
-        projectChanged = true;
-        needsNrwlWeb = true;
-      }
+  forEachExecutorOptions(
+    tree,
+    '@nrwl/angular:file-server',
+    (currentValue, project, target, configuration) => {
```

Comment:
We are not updating options and configurations. Using `forEachExecutorOptions` would needlessly execute the callback for every configuration and once for the options, but we only need it once per target.

We could have a different utility like `forEachTargetWithExecutor` for these use cases, but while we don't have it, it's better here to just change the existing code like:

```diff
- for (const [targetName, target] of Object.entries(project.targets)) {
+ for (const [targetName, target] of Object.entries(project.targets ?? {})) {
```

Reply:
I concur. I had originally wrote this with forEachExecutorOptions, but as Leo has mentioned, it runs needlessly.

Reply:
Sure, we can do that.